### PR TITLE
Hooks release

### PR DESCRIPTION
**Pull request by k-kryukov**
Release v1.0

### CODE REVIEW

File: commit-msg [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    commit_msg = "".join(["" if cur_s.startswith('#') else cur_s for cur_s in commit_msg.split('\n')])
+
+    if not re.fullmatch(regexStor, commit_msg) and not re.fullmatch(regexFix, commit_msg): # DOESNT MATCH
+        warning_text = open(".git/hooks/pre-commit-message/pre-commit-message-warnings").read()
+        help_text = open(".git/hooks/pre-commit-message/pre-commit-message-helper").read()
+        print(yellow_color + warning_text + color_off)
+        print(green_color + "HINT: " + color_off + help_text)
+    else: # MATCH
+        print(green_color + "Good Commit!" + color_off)
+# STOR-"numbers": ...
```
Comment (alexeyandriyash) [inline commit-msg:32]: looks like some docs, candidate to move into readme file

File: prepare-commit-msg [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    branch_fix = re.search('FIX', branch)
+
+    prev_content = f.readlines()
+    f.seek(0, 0)
+    lines = ["" if cur_s.startswith('#') else cur_s for cur_s in prev_content] # worthful part of message
+    whole_file = "".join(lines)
+    stor_commit = re.search('STOR-[0-9]{5}', whole_file)
+    fix_commit = re.search('FIX', whole_file)
+
+    #print(whole_file)
```
Comment (alexeyandriyash) [inline prepare-commit-msg:32]: ?

File: prepare-commit-msg [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+with open(commit_msg_filepath, "r+") as f:
+    branch_stor = re.search('STOR-[0-9]{5}', branch)
+    branch_fix = re.search('FIX', branch)
+
+    prev_content = f.readlines()
+    f.seek(0, 0)
+    lines = ["" if cur_s.startswith('#') else cur_s for cur_s in prev_content] # worthful part of message
+    whole_file = "".join(lines)
+    stor_commit = re.search('STOR-[0-9]{5}', whole_file)
+    fix_commit = re.search('FIX', whole_file)
```
Comment (alexeyandriyash) [inline prepare-commit-msg:30]: same

File: prepare-commit-msg [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        f.write("".join(lines))
+        f.write("\n#Don't forget to start with STOR-*****\n")
+        f.write("".join( ["" if not cur_s.startswith('#') else cur_s for cur_s in prev_content] ))
+        exit(0)
+    elif fix_commit:
+        print(green_color + "Updated your commit by adding STOR-number(or FIX) at the beginning" + color_off)
+        f.write("FIX: ")
+        f.write("".join(lines))
+        f.write("\n#Don't forget to start with FIX\n")
+        f.write("".join( ["" if not cur_s.startswith('#') else cur_s for cur_s in prev_content] ))
```
Comment (alexeyandriyash) [inline prepare-commit-msg:49]: same text as at 38 line

File: prepare-commit-msg [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    branch = "1234"
+
+green_color = "\033[1;32m"
+color_off = "\033[0m"
+
+print() # TO SEPARATE BLOCKS OF COMMITS
+
+with open(commit_msg_filepath, "r+") as f:
+    branch_stor = re.search('STOR-[0-9]{5}', branch)
+    branch_fix = re.search('FIX', branch)
```
Comment (alexeyandriyash) [inline prepare-commit-msg:23]: better to use commit prefix

File: prepare-commit-msg [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    whole_file = "".join(lines)
+    stor_commit = re.search('STOR-[0-9]{5}', whole_file)
+    fix_commit = re.search('FIX', whole_file)
+
+    #print(whole_file)
+    #print("STOR-[0-9]{5}")
+    if re.match('STOR-[0-9]{5}', whole_file) or whole_file.startswith('FIX'):
+        exit(0)
+
+    if stor_commit:
```
Comment (alexeyandriyash) [inline prepare-commit-msg:37]: you can eliminate this "if chain" by using commit_prefix

File: prepare-commit-msg [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        exit(0)
+    elif fix_commit:
+        print(green_color + "Updated your commit by adding STOR-number(or FIX) at the beginning" + color_off)
+        f.write("FIX: ")
+        f.write("".join(lines))
+        f.write("\n#Don't forget to start with FIX\n")
+        f.write("".join( ["" if not cur_s.startswith('#') else cur_s for cur_s in prev_content] ))
+        exit(0)
+
+    if lines == prev_content: # there is "-m" key or message is not empty
```
Comment (alexeyandriyash) [inline prepare-commit-msg:52]: previous commit message file content? prev_msg_content

File: commit-msg (hunk 1 above)
Comment (k-kryukov) [reply commit-msg:32]: done

File: prepare-commit-msg (hunk 7 above)
Comment (k-kryukov) [reply prepare-commit-msg:52]: done

File: prepare-commit-msg (hunk 6 above)
Comment (k-kryukov) [reply prepare-commit-msg:37]: done

File: prepare-commit-msg (hunk 5 above)
Comment (k-kryukov) [reply prepare-commit-msg:23]: done

File: prepare-commit-msg (hunk 3 above)
Comment (k-kryukov) [reply prepare-commit-msg:30]: done

File: README.md [hunk 8]
```diff
@@ -1 +1,27 @@
...
+2) "FIX"...
+If changes are not big, you can use FIX prefix in commit message
+
+2. commit-msg
+Allows to check whether commit message matches the template "STOR-***** ..." or "FIX ..."
+
+3. pre-commit
+Executes couple of utilities which search for mistakes in your code
+Config file may be changed if you want to expand executing utilities
+
```
Comment (alexeyandriyash) [inline README.md:26]: markdown lint needed

File: README.md [hunk 9]
```diff
@@ -1 +1,27 @@
...
+To deactive particular utility in pre_commit hook change corresponding field in utilities-config
+To deactivate all hooks just delete ".git/hooks"
+
+List of hooks:
+1. prepare-commit-msg
+Allows to correct commit message if it's possible
+Commit message requirements: commit message must begin like 
+1) "STOR-*story number*"... (story number usually consists of 5 digits)
+If it is possible, hook insert prefix to the beginning of commit message
+(for example, if branch name contains story number or story number is placed not at the beginning)
```
Comment (alexeyandriyash) [inline README.md:16]: please move to the first line of hook description

File: README.md (hunk 8 above)
Comment (alexeyandriyash) [inline README.md:26]: please describe pre-commit utilities and how they could be configured

Comment (alexeyandriyash) [reply README.md:26]: pep8+

File: git_hooks_source_files/pre-commit [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+def dot_pos(s : str):
+    dot_positions_list = []
+
+    for i in range(len(s)):
+        if s[i] == '.':
+            dot_positions_list.append(i)
+    return dot_positions_list
+
+def get_index_files():
+    #os.chdir('../../')
```
Comment (alexeyandriyash) [inline git_hooks_source_files/pre-commit:16]: typo

**Comment by alexeyandriyash** [comment]
LGTM

**Comment by alexeyandriyash** [comment]
Could you please add test scripts :-)

**Pull request merged**